test/simplify_algebra_test.cpp

```diff
@@ -1847,16 +1903,19 @@ TEST_CASE(reorder_reshape_slice)
         return m2;
     };
 
-    auto test = [&](std::size_t batch_size) {
-        auto m1 = create_m1(batch_size);
+    auto test = [&](std::size_t batch_size, bool input_transpose) {
+        auto m1 = create_m1(batch_size, input_transpose);
         run_pass(m1);
-        auto m2 = create_m2(batch_size);
+        auto m2 = create_m2(batch_size, input_transpose);
         EXPECT(m1.sort() == m2.sort());
     };
 
-    test(1);
-    test(4);
-    test(8);
+    test(1, true); // check if contiguous is added as necessary if input is transposed
+    test(1, false);
+    test(4, true);
+    test(4, false);
+    test(8, true);
+    test(8, false);
```

Comment:
I think it would be better to make this a template function and register each test case with `TEST_CASE_REGISTER`. Like what we do here: https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/test/float_equal.cpp#L71. This makes debugging easier as we can just run one test case.

Reply:
Done